### PR TITLE
ci:tdx: Use an ITA key for TDX

### PR DESCRIPTION
**Pull request by GabyCT**
This PR replaces the use of DCAP to use an ITA key for Kata CoCoTDX CI.

### CODE REVIEW

**Comment by GabyCT** [comment]
@fidencio this is the first draft of the ita change

**Comment by fidencio** [comment]
@GabyCT, I think this should work, now we are blocked on getting access to an ita key for the ci.

**Comment by GabyCT** [comment]
seems like attestation is not working from what I can see https://github.com/kata-containers/kata-containers/actions/runs/10949974759/job/30461032947?pr=10305
```
1..3
not ok 1 Test that creating a container from an authenticated image, with correct credentials works
# (from function `k8s_create_pod' in file lib.sh, line 63,
#  in test file k8s-guest-pull-image-authenticated.bats, line 75)
#   `k8s_create_pod "${kata_pod}"' failed
#       Reason:    CrashLoopBackOff
#     Last State:  Terminated
#       Reason:    StartError
#       Message:   failed to create containerd task: failed to create shim task: failed to get registry authentication credentials: ttrpc request error
# Caused by:
#     rpc status: Status { code: INTERNAL, message: "[CDH] [ERROR]: Get Resource failed", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }

2024-09-21T02:53:59.5147575Z # Sep 21 02:52:02 b49691c22b14 kata[250930]: time="2024-09-21T02:52:02.920561858Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f/console.sock name=containerd-shim-v2 pid=250930 sandbox=cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f source=virtcontainers subsystem=sandbox vmconsole="\x1b[90m[\x1b[0m2024-09-21T02:52:02Z \x1b[33mWARN \x1b[0m kbs_protocol::client::rcar_client\x1b[90m]\x1b[0m RCAR handshake failed: RCAR handshake failed: KBS attest unauthorized, Error Info: ErrorInformation { error_type: \"https://github.com/confidential-containers/kbs/errors/AttestationFailed\", detail: \"Attestation failed: Failed to deserialize attestation error response\\n\\nCaused by:\\n    0: error decoding response body\\n    1: missing field `error` at line 1 column 44\" }, retry 2..."
2024-09-21T02:53:59.5261038Z # Sep 21 02:52:09 b49691c22b14 kata[250930]: time="2024-09-21T02:52:09.927674906Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f/console.sock name=containerd-shim-v2 pid=250930 sandbox=cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f source=virtcontainers subsystem=sandbox vmconsole="    get resource failed"
2024-09-21T02:53:59.5264963Z # Sep 21 02:52:09 b49691c22b14 kata[250930]: time="2024-09-21T02:52:09.927740297Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f/console.sock name=containerd-shim-v2 pid=250930 sandbox=cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f source=virtcontainers subsystem=sandbox vmconsole="    Caused by: Kbs client error: get resource failed: get token failed: Attestation Agent token provider error: cal ttrpc failed: rpc status: Status { code: INTERNAL, message: \"[ERROR:attestation-agent] AA-KBC get token failed\", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }"
2024-09-21T02:53:59.5270672Z # Sep 21 02:52:09 b49691c22b14 kata[250930]: time="2024-09-21T02:52:09.930202138Z" level=error msg="createContainer failed" error="rpc error: code = Internal desc = failed to get registry authentication credentials: ttrpc request error\n\nCaused by:\n    rpc status: Status { code: INTERNAL, message: \"[CDH] [ERROR]: Get Resource failed\", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }\n\nStack backtrace:\n   0: <unknown>\n   1: <unknown>\n   2: <unknown>\n   3: <unknown>\n   4: <unknown>\n   5: <unknown>\n   6: <unknown>\n   7: <unknown>\n   8: <unknown>\n   9: <unknown>\n  10: <unknown>\n  11: <unknown>\n  12: <unknown>\n  13: <unknown>\n  14: <unknown>\n  15: <unknown>\n  16: <unknown>\n\nStack backtrace:\n   0: <unknown>\n   1: <unknown>\n   2: <unknown>\n   3: <unknown>\n   4: <unknown>\n   5: <unknown>\n   6: <unknown>\n   7: <unknown>\n   8: <unknown>\n   9: <unknown>\n  10: <unknown>\n  11: <unknown>\n  12: <unknown>\n  13: <unknown>\n  14: <unknown>" name=containerd-shim-v2 pid=250930 sandbox=cb4b4b9644884a87f7fc3cea6bd5af2bbf8e8647b7bcf58a1952f370aacfc32f source=virtcontainers subsystem=kata_agent


not ok 2 Test that creating a container from an authenticated image, with incorrect credentials fails
2024-09-21T02:56:15.2442325Z # Sep 21 02:54:20 b49691c22b14 kata[253405]: time="2024-09-21T02:54:20.888836228Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/196962792d1d89cfffad5b7f4c622bcc3e3f3f96a5f9ccff5069e79e81face3a/console.sock name=containerd-shim-v2 pid=253405 sandbox=196962792d1d89cfffad5b7f4c622bcc3e3f3f96a5f9ccff5069e79e81face3a source=virtcontainers subsystem=sandbox vmconsole="\x1b[90m[\x1b[0m2024-09-21T02:54:20Z \x1b[33mWARN \x1b[0m kbs_protocol::client::rcar_client\x1b[90m]\x1b[0m RCAR handshake failed: RCAR handshake failed: KBS attest unauthorized, Error Info: ErrorInformation { error_type: \"https://github.com/confidential-containers/kbs/errors/AttestationFailed\", detail: \"Attestation failed: Failed to deserialize attestation error response\\n\\nCaused by:\\n    0: error decoding response body\\n    1: missing field `error` at line 1 column 44\" }, retry 4..."

not ok 1 Create a pod from an unsigned image, on an insecureAcceptAnything registry works
sandbox=80548207fbbba4b928798b649fedff5a576b9711a0f0c08e053e6d1fe37ed0c1 source=virtcontainers subsystem=sandbox vmconsole="    Caused by: Kbs client error: get resource failed: get token failed: Attestation Agent token provider error: cal ttrpc failed: rpc status: Status { code: INTERNAL, message: \"[ERROR:attestation-agent] AA-KBC get token failed\", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }"
2024-09-21T03:00:39.2342216Z # Sep 21 02:58:51 b49691c22b14 kata[258424]: time="2024-09-21T02:58:51.168902952Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/80548207fbbba4b928798b649fedff5a576b9711a0f0c08e053e6d1fe37ed0c1/console.sock name=containerd-shim-v2 pid=258424 

not ok 2 Create a pod from an unsigned image, on a 'restricted registry' is rejected
2024-09-21T03:02:54.2993144Z # Sep 21 03:01:03 b49691c22b14 kata[260937]: time="2024-09-21T03:01:03.831172697Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/ad2f51fb390dd1dbc6356f84374d5dfef7a4f1117f2f16835550f398dec4ac12/console.sock name=containerd-shim-v2 pid=260937 sandbox=ad2f51fb390dd1dbc6356f84374d5dfef7a4f1117f2f16835550f398dec4ac12 source=virtcontainers subsystem=sandbox vmconsole="    Caused by: Kbs client error: get resource failed: get token failed: Attestation Agent token provider error: cal ttrpc failed: rpc status: Status { code: INTERNAL, message: \"[ERROR:attestation-agent] AA-KBC get token failed\", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }"
2024-09-21T03:02:54.2996618Z # Sep 21 03:01:03 b49691c22b14 kata[260937]: time="2024-09-21T03:01:03.834809112Z" level=error msg="createContainer failed" error="rpc error: code = Internal desc = Security validate failed: ttrpc request error\n\nCaused by:\n    rpc status: Status { code: INTERNAL, message: \"[CDH] [ERROR]: Get Resource failed\", details: [], special_fields: SpecialFields { unknown_fields: UnknownFields { fields: None }, cached_size: CachedSize { size: 0 } } }\n\nStack backtrace:\n   0: <unknown>\n   1: <unknown>\n   2: <unknown>\n   3: <unknown>\n   4: <unknown>\n   5: <unknown>\n   6: <unknown>\n   7: <unknown>\n   8: <unknown>\n   9: <unknown>\n  10: <unknown>\n  11: <unknown>\n  12: <unknown>\n  13: <unknown>\n  14: <unknown>\n  15: <unknown>\n  16: <unknown>\n\nStack backtrace:\n   0: <unknown>\n   1: <unknown>\n   2: <unknown>\n   3: <unknown>\n   4: <unknown>\n   5: <unknown>\n   6: <unknown>\n   7: <unknown>\n   8: <unknown>\n   9: <unknown>\n  10: <unknown>\n  11: <unknown>\n  12: <unknown>\n  13: <unknown>\n  14: <unknown>" name=containerd-shim-v2 pid=260937 sandbox=ad2f51fb390dd1dbc6356f84374d5dfef7a4f1117f2f16835550f398dec4ac12 source=virtcontainers subsystem=kata_agent

not ok 3 Create a pod from a signed image, on a 'restricted registry' is successful
2024-09-21T03:05:08.1471823Z # Sep 21 03:03:26 b49691c22b14 kata[263708]: time="2024-09-21T03:03:26.333401673Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/b16027aaf2572d7cd1a73a7e2c3c31da097a790b30b35dc7a12eae1bc59dba0a/console.sock name=containerd-shim-v2 pid=263708 sandbox=b16027aaf2572d7cd1a73a7e2c3c31da097a790b30b35dc7a12eae1bc59dba0a source=virtcontainers subsystem=sandbox vmconsole="\x1b[90m[\x1b[0m2024-09-21T03:03:24Z \x1b[33mWARN \x1b[0m kbs_protocol::client::rcar_client\x1b[90m]\x1b[0m RCAR handshake failed: RCAR handshake failed: KBS attest unauthorized, Error Info: ErrorInformation { error_type: \"https://github.com/confidential-containers/kbs/errors/AttestationFailed\", detail: \"Attestation failed: Failed to deserialize attestation error response\\n\\nCaused by:\\n    0: error decoding response body\\n    1: missing field `error` at line 1 column 44\" }, retry 3..."

not ok 4 Create a pod from a signed image, on a 'restricted registry', but with the wrong key is rejected
\x1b[0m attester::tdx\x1b[90m]\x1b[0m Read AA Eventlog failed: Os { code: 2, kind: NotFound, message: \"No such file or directory\" }"
2024-09-21T03:07:25.6133385Z # Sep 21 03:05:39 b49691c22b14 kata[266145]: time="2024-09-21T03:05:39.514925067Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/90329ef001121dda4868b5aa05c4b18ad56955958211ccc18f177973d3bf704e/console.sock name=containerd-shim-v2 pid=266145 sandbox=90329ef001121dda4868b5aa05c4b18ad56955958211ccc18f177973d3bf704e source=virtcontainers subsystem=sandbox vmconsole="\x1b[90m[\x1b[0m2024-09-21T03:05:37Z \x1b[33mWARN \x1b[0m kbs_protocol::client::rcar_client\x1b[90m]\x1b[0m RCAR handshake failed: RCAR handshake failed: KBS attest unauthorized, Error Info: ErrorInformation { error_type: \"https://github.com/confidential-containers/kbs/errors/AttestationFailed\", detail: \"Attestation failed: Failed to deserialize attestation error response\\n\\nCaused by:\\n    0: error decoding response body\\n    1: missing field `error` at line 1 column 44\" }, retry 3..."

not ok 1 Get CDH resource
21T03:08:04.469434625Z\",\"pid\":\"126\",\"source\":\"agent\",\"version\":\"0.1.0\",\"subsystem\":\"rpc\",\"name\":\"kata-agent\"}"
2024-09-21T03:08:12.6909680Z # Sep 21 03:08:06 b49691c22b14 kata[270389]: time="2024-09-21T03:08:06.204802345Z" level=debug msg="reading guest console" console-protocol=unix console-url=/run/vc/vm/f92dd571819a7369ae0559547bd99f8a823d89fd9fe94080efcb2019dedb708f/console.sock name=containerd-shim-v2 pid=270389 sandbox=f92dd571819a7369ae0559547bd99f8a823d89fd9fe94080efcb2019dedb708f source=virtcontainers subsystem=sandbox vmconsole="\x1b[90m[\x1b[0m2024-09-21T03:08:04Z \x1b[33mWARN \x1b[0m kbs_protocol::client::rcar_client\x1b[90m]\x1b[0m RCAR handshake failed: RCAR handshake failed: KBS attest unauthorized, Error Info: ErrorInformation { error_type: \"https://github.com/confidential-containers/kbs/errors/AttestationFailed\", detail: \"Attestation failed: Failed to deserialize attestation error response\\n\\nCaused by:\\n    0: error decoding response body\\n    1: missing field `error` at line 1 column 44\" }, retry 3..."
```

File: tests/integration/kubernetes/confidential_kbs.sh
Comment (dborquez) [inline tests/integration/kubernetes/confidential_kbs.sh:326]: The string `tBfd5kKX2x9ahbodKV1` is always constant?
Could you add a comment mentioning which `property` of the KBS config file is being affected by this line?
[KBS Configuration File]: https://github.com/confidential-containers/trustee/blob/main/kbs/docs/config.md

Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh:326]: The string is always constant.

However, the documentation you pointed out doesn't apply for ITA / ITTS.  The `sed` replaces https://github.com/confidential-containers/trustee/blob/3b7e73d760feb8650db7c88b19c5e1bf9968d071/kbs/config/kubernetes/ita/kbs-config.toml#L13

I'd prefer adding this as a follow-up (instead of adding it here and then having to wait a few more hours for the CI to run again), but it can only be added after we expand the KBS documentation for the kustomizations.

Can you open an issue on Trustee to add docs for the kustomizations?  Then we do it there, and reflect the changes here as well.

Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh:326]: Actually there's doc for ITA: https://github.com/confidential-containers/trustee/blob/main/kbs/docs/config.md#intel-trust-authority-formerly-known-as-amber

I'd still prefer adding this as a follow-up, just for the sake of not waiting for 1~2 hours before having this one merged.

@GabyCT, can you work on the follow-up once this one is merged?

Comment (fidencio) [reply tests/integration/kubernetes/confidential_kbs.sh:326]: Argh, as the metrics CI is failing, let me do this change now.

**Review by dborquez — APPROVED** [review]
LGTM, thank you @GabyCT

**Review by dborquez — APPROVED** [review]
LGTM, thank you @GabyCT

**Review by chavafg — APPROVED** [review]

**Comment by fidencio** [comment]
Tests which require attestation are all passing. There are still failures happening every now and then which will be solved by https://github.com/kata-containers/kata-containers/pull/10353

**Review by amshinde — APPROVED** [review]

**Pull request merged**
